tests/lib/devicetree/devices/libraries.devicetree.devices: build failure (bl5340_dvk_cpuapp)

**Describe the bug**

```
zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m33  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,--fatal-warnings && cd /__w/zephyr/zephyr/twister-out/bl5340_dvk_cpuapp/tests/lib/devicetree/devices/libraries.devicetree.devices/zephyr && /usr/local/bin/cmake -E echo
/opt/toolchains/zephyr-sdk-0.15.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/i2c/libdrivers__i2c.a(i2c_test.c.obj):/__w/zephyr/zephyr/drivers/i2c/i2c_test.c:46: multiple definition of `__device_dts_ord_150'; app/libapp.a(main.c.obj):/__w/zephyr/zephyr/tests/lib/devicetree/devices/src/main.c:39: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**To Reproduce**

```
west build -b bl5340_dvk_cpuapp tests/lib/devicetree/devices -p
```

**Expected behavior**

`tests/lib/devicetree/devices` should only be run for `native_posix` on CI, it doesn't make sense to run this kind of tests ("unit") on other platforms.

**Impact**

Blocking CI

**Logs and console output**

See above

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.15.0
 - Commit SHA or Version used 6755999bb3f6b760347a57502715c40cf8881636

**Additional context**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/3272340917/jobs/5383270284

--- a/tests/lib/devicetree/devices/testcase.yaml
+++ b/tests/lib/devicetree/devices/testcase.yaml
@@ -5,4 +5,5 @@ tests:
     # will mostly likely be the fastest.
     integration_platforms:
       - native_posix
-    platform_exclude: hsdk hsdk_2cores thingy52_nrf52832 bbc_microbit bbc_microbit_v2
+    platform_exclude: hsdk hsdk_2cores thingy52_nrf52832 bbc_microbit bbc_microbit_v2 bt610
+       bl5340_dvk_cpuapp bl5340_dvk_cpuapp_ns

